src/Temporalio/Worker/Tuning/FixedSizeSlotSupplier.cs

```diff
@@ -0,0 +1,8 @@
+namespace Temporalio.Worker.Tuning
+{
+    /// <summary>
+    /// A slot supplier that will only ever issue at most a fixed number of slots.
+    /// </summary>
+    /// <param name="NumSlots">The maximum number of slots that will ever be issued.</param>
+    public sealed record FixedSizeSlotSupplier(uint NumSlots) : ISlotSupplier;
```

Comment:
```suggestion
    public sealed record FixedSizeSlotSupplier(int SlotCount) : ISlotSupplier;
```

Besides the int/uint thing, I think .NET would prefer either `Count` suffix or just `Slots` over "num".